source/sigv4.c

```diff
@@ -292,20 +306,46 @@ static SigV4Status_t appendCanonicalizedHeaders( size_t headerCount,
  * in the canonical form.
  * @param[in,out] canonicalRequest Struct to maintain intermediary buffer
  * and state of canonicalization.
+ *
+ * @return Following statuses will be returned by the function:
+ * #SigV4Success if headers are successfully added to the canonical request.
+ * #SigV4InsufficientMemory if buffer remaining cannot accommodate all the provided headers.
  */
-static SigV4Status_t writeSignedHeaderToString( size_t headerIndex,
-                                                uint32_t flags,
-                                                CanonicalContext_t * canonicalRequest );
+static SigV4Status_t writeSignedHeaderToCanonicalRequest( size_t headerIndex,
+                                                          uint32_t flags,
+                                                          CanonicalContext_t * canonicalRequest );
 
 /**
  * @brief Write canonical headers to the buffer provided.
  *
  * @param[in] headerIndex Index of header to write to buffer.
  * @param[in,out] canonicalRequest Struct to maintain intermediary buffer
  * and state of canonicalization.
+ *
+ * @return Following statuses will be returned by the function:
+ * #SigV4Success if headers are successfully added to the canonical request.
+ * #SigV4InsufficientMemory if buffer remaining cannot accommodate all the provided headers.
+ */
+static SigV4Status_t writeCanonicalHeaderToCanonicalRequest( size_t headerIndex,
+                                                             CanonicalContext_t * canonicalRequest );
+
+/**
+ * @brief Helper function to check whether the current character is a
+ * space that is either trailing, leading, or having multiple occurrences
+ * between characters in a string.
+ *
+ * @param[in] value String to be trimmed.
```

Comment:
```suggestion
 * @param[in] value Header value to be trimmed.
```

Reply:
It can be header value or key so should not we keep it more generalized.

Reply:
Would be better to call it "Header value or key string"